CMakeLists.txt

```diff
@@ -0,0 +1,102 @@
+cmake_minimum_required(VERSION 3.15)
+project(NetHack VERSION 3.7)
+
+if(CMAKE_BUILD_TYPE)
+  message("Build type: ${CMAKE_BUILD_TYPE}.")
+endif()
+
+message(STATUS "Building NetHack version: ${NetHack_VERSION}.")
+
+set(HACKDIR
+    "$ENV{HOME}/nethackdir.37"
+    CACHE STRING "Configuration files for nethack")
+
+message(STATUS "HACKDIR set to: ${HACKDIR}")
+
+include(ExternalProject)
+
+# Playground vars
+set(VARDIR ${HACKDIR})
+set(INSTDIR ${HACKDIR})
+
+add_compile_definitions(
+  GCC_WARN
+  NOCLIPPING
+  NOMAIL
+  NOTPARMDECL
+  AVOID_WIN_IOCTL
+  SCORE_ON_BOTL
+  # ANSI_DEFAULT NO_SIGNAL
+  HACKDIR="${HACKDIR}"
+  DEFAULT_WINDOW_SYS="tty"
+  DLB)
+
+set(NH_SRC ${NetHack_SOURCE_DIR}/src)
+set(NH_INC ${NetHack_SOURCE_DIR}/include)
+set(NH_DAT ${NetHack_SOURCE_DIR}/dat)
+set(NH_UTIL ${NetHack_SOURCE_DIR}/util)
+set(NH_DOC ${NetHack_SOURCE_DIR}/doc)
+
+set(NH_SRC_GEN ${NetHack_BINARY_DIR}/src)
+set(NH_INC_GEN ${NetHack_BINARY_DIR}/include)
+set(NH_DAT_GEN ${NetHack_BINARY_DIR}/dat)
+set(NH_UTIL_GEN ${NetHack_BINARY_DIR}/util)
+
+# Lua.
+set(LUA_VERSION 5.4.3)
+
+ExternalProject_Add(
+  fetch-lua
+  PREFIX "lua"
+  URL http://www.lua.org/ftp/lua-${LUA_VERSION}.tar.gz
+  URL_HASH
+    SHA512=3a1a3ee8694b72b4ec9d3ce76705fe179328294353604ca950c53f41b41161b449877d43318ef4501fee44ecbd6c83314ce7468d7425ba9b2903c9c32a28bbc0
+  BUILD_IN_SOURCE TRUE
+  CONFIGURE_COMMAND ""
+  BUILD_COMMAND make
+  INSTALL_COMMAND ""
+  STEP_TARGETS build)
+
+ExternalProject_Get_Property(fetch-lua BINARY_DIR)
+set(LUA_INCLUDE ${BINARY_DIR}/src)
+set(TOPLUALIB ${BINARY_DIR}/src/liblua.a)
+
+add_custom_command(
+  OUTPUT ${NH_INC_GEN}/nhlua.h
+  COMMAND ${CMAKE_COMMAND} -E echo "/* nhlua.h - generated by top Makefile */" >
+          ${NH_INC_GEN}/nhlua.h
+  COMMAND ${CMAKE_COMMAND} -E echo "#include \"lua.h\"" >> ${NH_INC_GEN}/nhlua.h
+  COMMAND ${CMAKE_COMMAND} -E echo "#include \"lualib.h\"" >>
+          ${NH_INC_GEN}/nhlua.h
+  COMMAND ${CMAKE_COMMAND} -E echo "#include \"lauxlib.h\"" >>
+          ${NH_INC_GEN}/nhlua.h
+  COMMAND ${CMAKE_COMMAND} -E echo "'/*nhlua.h*/" >> ${NH_INC_GEN}/nhlua.h
+  VERBATIM)
+
+add_custom_target(lua_support DEPENDS fetch-lua-build ${NH_INC_GEN}/nhlua.h)
+
+add_subdirectory(util)
+add_subdirectory(dat)
+
+file(
+  GLOB
+  NETHACK_SRC
```

Comment:
I've seen this disclaimer. I was going to push back that NetHack's current Makefiles do the same, but that's not true (that argument would apply for [this glob](https://github.com/heiner/NetHack/blob/NetHack-3.7-cmake/dat/CMakeLists.txt#L25)). Instead, `Makefile.src` actually lists all files in `src/`. I'm happy to update the PR to do the same.

Reply:
Updated the PR to list the sources w/o `GLOB`. This makes the top `CMakeLists.txt` a bit longer; one could consider moving parts of it into a `src/CMakeLists.txt`.

Reply:
It's cumbersome but I always list the sources. I've not used the `CONFIGURE_DEPENDS` feature but I think I read somewhere that it's reliable for Ninja and Make.

With just plain `GLOB` everyone needs to manually rerun `cmake` anytime someone adds or removes a source file, which isn't great. If they forget they either get an error or I think in some cases it can look like it succeeds but result in broken programs.

Reply:
As I said the PR is updated to remove this GLOB. Can this conversation be "resolved"?

Reply:
I can not find the button otherwise I would have done so!

Docs here say that you can only resolve a conversation when you're the one that opened the pull request or if you have write access to the repo. https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/commenting-on-a-pull-request#resolving-conversations

Can you resolve the conversation? Otherwise I'll just delete the root comment